### DISCUSSION TITLE
How to feed an image folder to ControlNet in Web API mode?

### DISCUSSION THREAD

**Discussion by uk9921** [Mikubill/sd-webui-controlnet#2207] [Q&A] [answered]
I have learned the basic usage of using **Web API** to call ControlNet by passing the `base64.b64encode` of an image as the value of the `"image" `key. However, I am unsure how to feed an image folder to ControlNet.
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/20817396/1f9f9da5-103b-4a60-a931-29b68d01db9e)
I have noticed that there doesn't seem to be any discussion on this issue. Does anyone have any suggestions?

**Comment by huchenlei** [comment, marked as answer]
If you are calling A1111 API, you can easily write up a loop to achieve this:

```python
for image_file in directory:
   image = read_image(image_file)
   result = call_api(image, ...)
```

**Reply by uk9921** [reply]
I modified the control_unit.batch_images to avoid redundant API calls, but your method is indeed very concise, haha.
